docs/src/details/common-widget-features/cache_framework/api_reference.rst

```diff
@@ -0,0 +1,586 @@
+.. _cache_framework_api_reference:
+
+API Reference
```

Comment:
We generate this from Doxygen comments. Please make sure that all functions have proper comments and add the auto-generated API reference. @vwheeler63 can help with how to add the API from specific headers.